hmpps-sqs-spring-boot-autoconfigure/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsQueueService.kt

```diff
@@ -40,10 +40,11 @@ class HmppsQueueService(
       .map { retryDlqRequest -> retryDlqMessages(retryDlqRequest) }
 
   private fun HmppsQueue.retryDlqMessages(): RetryDlqResult {
-    val messageCount = sqsDlqClient.countMessagesOnQueue(dlqUrl)
+    if (sqsDlqClient == null || dlqUrl == null) return RetryDlqResult(0, mutableListOf<Message>())
+    val messageCount = sqsDlqClient.countMessagesOnQueue(dlqUrl!!)
     val messages = mutableListOf<Message>()
     repeat(messageCount) {
-      sqsDlqClient.receiveMessage(ReceiveMessageRequest(dlqUrl).withMaxNumberOfMessages(1)).messages.firstOrNull()
+      sqsDlqClient!!.receiveMessage(ReceiveMessageRequest(dlqUrl).withMaxNumberOfMessages(1)).messages.firstOrNull()
```

Comment:
I don't think you need the bangbangs here - once you have null checked them on line 43 they should be smart casted to non-nullables.

Reply:
good spot - I missed that one